controllers/vspherevm_controller.go

```diff
@@ -363,16 +364,20 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	return reconcile.Result{}, nil
 }
 
+// isWaitingForStaticIPAllocation checks whether the VM should wait for a static IP
+// to be allocated.
+// It checks the state of both DHCP4 and DHCP6 for all the network devices and if
+// any static IP addresses are specified.
```

Comment:
Am just wondering, when one of the NIC needs static IP, If the vm got created without the static IP, how will the static IP to be set into the VM?  Do we expect some additional agent running within the VM to configure the additional NIC?

Reply:
Hey @jzhoucliqr , I looked at the proposal [here](https://docs.google.com/document/d/10I-HH67_xBk73pazhNmbJjOvtPgq9Uk6ZrD3vj06srs/edit#heading=h.iame4w8fq7my) and the idea was that we would be using an external IPAM for IP allocation and this address allocation logic is not to be included in the CAPV codebase. Hence made the changes in this way. Does that answer your question?

Reply:
Thanks @srm09 . The allocation is done by external IPAM, but to configure the IP within the VM is done by cloud-init, and we need to pass the IP info into the VM using guestinfo.metadata. So that means we'll need to wait for all IPs to be allocated then clone the VM, and set the correct static IP info.

Reply:
just took a look at this, @jzhoucliqr is correct we need to have the static IPs in place before creating the VM. I think we can close this PR

Reply:
Thanks @yastij and @jzhoucliqr for providing additional context